include/tvm/relay/pass.h

```diff
@@ -2,18 +2,202 @@
  *  Copyright (c) 2018 by Contributors
  * \file tvm/relay/pass.h
  * \brief The set of Relay passes written in C++.
+ *
+ * This file also implements a pass manager. The pass manager manages a sequence
+ * of Relay-to-Relay transformation passes over a particlar unit of AST. The
+ * design is largely inspired from LLVM's pass manager.
+ *
+ * The responsibilities of a pass manager usually involves:
+ *  - Organizing the execution order of optimization passes though not
+ * necessarily in the optimal sequence.
+ *  - Collecting required analysis information and keep them up-to-date.
+ *  - Reducing the effort required to implement new passes for compiler
+ * developers, etc.
+ *
+ * TODO(jroesch, zhiics): We are currently using a very simple design for the
```

Comment:
move todo to the cc file. The dependency tracking is a feature of SequentialPass and has nothing to do with the interface